compiler/rustc_privacy/src/lib.rs

```diff
@@ -636,8 +636,7 @@ impl<'tcx> EmbargoVisitor<'tcx> {
 impl<'tcx> Visitor<'tcx> for EmbargoVisitor<'tcx> {
     fn visit_item(&mut self, item: &'tcx hir::Item<'tcx>) {
         if self.impl_trait_pass
-            && let hir::ItemKind::OpaqueTy(opaque) = item.kind
-            && !opaque.in_trait
```

Comment:
I'm not sure why we were opting out `in_trait` from the embargo visitor, but I didn't see any change to behavior here.